src/globus_sdk/experimental/tokenstorage_v2/file_adapters.py

```diff
@@ -0,0 +1,171 @@
+from __future__ import annotations
+
+import json
+import pathlib
+import typing as t
+
+from globus_sdk.experimental.tokenstorage_v2.base import FileAdapterV2
+from globus_sdk.services.auth import OAuthTokenResponse
+from globus_sdk.version import __version__
+
+
+class JSONFileAdapter(FileAdapterV2):
+    """
+    A storage adapter for storing tokens in JSON files.
+    """
+
+    # the version for the current data format used by the file adapter.
+    # 1.0 was used by ``SimpleJSONFileAdapter``. If ``JSONFileAdapter`` is
+    # pointed at storage used by a ``SimpleJSONFileAdapter` it will be converted to 2.0
+    # and no longer usable by ``SimpleJSONFileAdapter``.
+    format_version = "2.0"
+
+    # the supported versions (data not in these versions causes an error)
+    supported_versions = ("1.0", "2.0")
+
+    def __init__(self, filename: pathlib.Path | str, *, namespace: str = "DEFAULT"):
+        """
+        :param filename: the name of the file to write to and read from
+        :param namespace: A user-supplied namespace for partitioning token data
+        """
+        self.filename = str(filename)
+        super().__init__(namespace=namespace)
+
+    def _raw_load(self) -> dict[str, t.Any]:
+        """
+        Load the file contents as JSON and return the resulting dict
+        object. If a dict is not found, raises an error.
+        """
+        with open(self.filename, encoding="utf-8") as f:
+            val = json.load(f)
+        if not isinstance(val, dict):
+            raise ValueError("reading from json file got non-dict data")
+        return val
+
+    def _handle_formats(self, read_data: dict[str, t.Any]) -> dict[str, t.Any]:
+        """Handle older data formats supported by this class
+
+        if the data is not in a known/recognized format, this will error
+        otherwise, reshape the data to the current supported format and return it
+        """
+        format_version = read_data.get("format_version")
+        if format_version not in self.supported_versions:
+            raise ValueError(
+                f"cannot store data using SimpleJSONFileAdapter({self.filename} "
+                "existing data file is in an unknown format "
+                f"(format_version={format_version})"
+            )
+
+        # 1.0 data was stored under a "by_rs" key without namespaces, to upgrade we
+        # move everything under the "DEFAULT" key and remove the "by_rs" key.
+        if format_version == "1.0":
+            read_data = {
+                "data": {
+                    "DEFAULT": read_data["by_rs"],
+                },
+                "format_version": self.format_version,
+                "globus-sdk.version": __version__,
```

Comment:
Why the difference in delimiter here? `_` vs `-`

Reply:
I think this is the only comment that didn't get addressed in my last commit/rebase.

Essentially I just wanted to keep the value used by `SimpleJSONStorageAdapter` so there was one less thing for users to worry about when migrating.

As for why it was like this originally :shrug: